core/templates/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -432,95 +470,124 @@ angular.module('oppia').directive('questionsList', [
-                groupedSkillSummaries: () => groupedSkillSummaries,
-                misconceptionsBySkill: () => misconceptionsBySkill,
-                newQuestionIsBeingCreated: () => newQuestionIsBeingCreated,
-                question: () => question,
-                questionId: () => questionId,
-                questionStateData: () => questionStateData,
-                rubrics: () => rubrics,
-                skillNames: () => skillNames
+                skillsInSameTopicCount: () => skillsInSameTopicCount,
+                sortedSkillSummaries: () => sortedSkillSummaries,
+                categorizedSkills: () => ctrl.getSkillsCategorizedByTopics,
+                allowSkillsFromOtherTopics: () => allowSkillsFromOtherTopics,
+                untriagedSkillSummaries: () => ctrl.getUntriagedSkillSummaries
               },
-              controller: 'QuestionEditorModalController',
-            }).result.then(function(modalObject) {
-              ContextService.resetImageSaveDestination();
-              $location.hash(null);
-              ctrl.editorIsOpen = false;
-              if (modalObject.skillLinkageModificationsArray.length > 0) {
-                EditableQuestionBackendApiService.editQuestionSkillLinks(
-                  questionId, modalObject.skillLinkageModificationsArray,
-                  questionDifficulty).then(
-                  data => {
-                    $timeout(function() {
-                      QuestionsListService.resetPageNumber();
-                      _reInitializeSelectedSkillIds();
-                      ctrl.getQuestionSummariesAsync(
-                        ctrl.selectedSkillIds, true, true
-                      );
-                      ctrl.saveAndPublishQuestion(modalObject.commitMessage);
-                    }, 500);
-                  });
-              } else {
-                ContextService.resetImageSaveDestination();
-                ctrl.saveAndPublishQuestion(modalObject.commitMessage);
+              controller: 'SelectSkillModalController',
+              windowClass: 'skill-select-modal',
+              size: 'xl'
+            }).result.then(function(summary) {
+              for (var idx in ctrl.associatedSkillSummaries) {
+                if (
+                  ctrl.associatedSkillSummaries[idx].getId() ===
+                    summary.id) {
+                  AlertsService.addInfoMessage(
+                    'Skill already linked to question');
+                  return;
+                }
               }
+
+              ctrl.associatedSkillSummaries.push(
+                ShortSkillSummaryObjectFactory.create(
+                  summary.id, summary.description));
+              ctrl.skillLinkageModificationsArray = [];
+              ctrl.skillLinkageModificationsArray.push({
+                id: summary.id,
+                task: 'add'
+              });
             }, function() {
-              ContextService.resetImageSaveDestination();
```

Comment:
Why delete these?

Reply:
These are not removed, just refactored. See line 576, 584.